src/NzbDrone.Core/Indexers/Definitions/Cardigann/CardigannRequestGenerator.cs

```diff
@@ -722,49 +722,60 @@ public async Task<HttpRequest> DownloadRequest(Uri link)
+                        try
+                        {
+                            var request = new HttpRequestBuilder(link.ToString())
+                                .SetCookies(Cookies ?? new Dictionary<string, string>())
+                                .SetHeaders(headers ?? new Dictionary<string, string>())
+                                .Build();
 
-                    request.AllowAutoRedirect = true;
+                            request.AllowAutoRedirect = true;
 
-                    var response = await HttpClient.ExecuteAsync(request);
+                            var response = await HttpClient.ExecuteAsync(request);
 
-                    var results = response.Content;
-                    var searchResultParser = new HtmlParser();
-                    var searchResultDocument = searchResultParser.ParseDocument(results);
-                    var downloadElement = searchResultDocument.QuerySelector(selector);
-                    if (downloadElement != null)
-                    {
-                        _logger.Debug(string.Format("CardigannIndexer ({0}): Download selector {1} matched:{2}", _definition.Id, selector, downloadElement.ToHtmlPretty()));
+                            var results = response.Content;
+                            var searchResultParser = new HtmlParser();
+                            var searchResultDocument = searchResultParser.ParseDocument(results);
+                            var downloadElement = searchResultDocument.QuerySelector(queryselector);
+                            if (downloadElement != null)
+                            {
+                                _logger.Debug(string.Format("CardigannIndexer ({0}): Download selector {1} matched:{2}", _definition.Id, queryselector, downloadElement.ToHtmlPretty()));
 
-                        var href = "";
-                        if (download.Attribute != null)
-                        {
-                            href = downloadElement.GetAttribute(download.Attribute);
-                            if (href == null)
+                                var href = "";
+                                if (selector.Attribute != null)
+                                {
+                                    href = downloadElement.GetAttribute(selector.Attribute);
+                                    if (href == null)
+                                    {
+                                        throw new Exception(string.Format("Attribute \"{0}\" is not set for element {1}", selector.Attribute, downloadElement.ToHtmlPretty()));
+                                    }
+                                }
+                                else
+                                {
+                                    href = downloadElement.TextContent;
+                                }
+
+                                href = ApplyFilters(href, download.Filters, variables);
+                                link = ResolvePath(href, link);
```

Comment:
We should likely return result here since we have a matching selector instead of letting the loop continue to the next item. Check out the Jackett code for ref on both logics.

Reply:
seems jackett tests out the link before returning it?
https://github.com/Jackett/Jackett/blob/7a27bb6eb4138466100185005fb0468649f4e741/src/Jackett.Common/Indexers/CardigannIndexer.cs#L1802

is that 1) something handled elsewhere and 2) if not, something we should handle?

or is it as simple as just 
return link;

assuming the latter for the moment